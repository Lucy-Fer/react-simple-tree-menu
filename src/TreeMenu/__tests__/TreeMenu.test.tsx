import * as React from 'react';
import { shallow } from 'enzyme';

import TreeViewMenu from '../index';

const mockData = {
  atd: {
    label: 'ATS Guide',
    url: 'ats',
    index: 1, // ATS Guide should be after Release Notes
  },
  releasenotes: {
    label: 'Release Notes',
    url: 'releasenotes',
    index: 0, // Release Notes should be first
    nodes: {
      'desktop-modeler': {
        label: 'Desktop Modeler',
        url: 'releasenotes/desktop-modeler',
        index: 0,
        nodes: {
          7: {
            label: '7',
            url: 'releasenotes/desktop-modeler/7',
            index: 0,
            nodes: {
              '7.0': {
                label: '7.0',
                url: 'releasenotes/desktop-modeler/7.0',
                index: 0,
              },
            },
          },
        },
      },
    },
  },
};

describe('TreeViewMenu', () => {
  it('should render the roots only', () => {
    const wrapper = shallow(
      <TreeViewMenu data={mockData} activeKey="releasenotes/desktop-modeler/7" />
    );

    expect(wrapper).toMatchSnapshot();
  });
});
