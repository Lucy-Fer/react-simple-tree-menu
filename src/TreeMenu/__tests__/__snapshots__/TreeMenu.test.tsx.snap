// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TreeViewMenu should render the roots only 1`] = `
<Fragment>
  <input
    onChange={[Function]}
    placeholder="Type and search"
    style={
      Object {
        "margin": ".5em",
        "paddingLeft": ".4em",
      }
    }
  />
  <ul
    style={
      Object {
        "listStyleType": "none",
        "paddingLeft": 0,
      }
    }
  >
    <li
      key="releasenotes"
      onClick={[Function]}
      style={
        Object {
          "background": "none",
          "color": "#333",
          "cursor": "pointer",
          "paddingLeft": "2.25rem",
        }
      }
    >
      <div
        style={
          Object {
            "height": "1rem",
            "left": "1rem",
            "position": "absolute",
            "width": "1rem",
          }
        }
      >
        <ToggleIcon
          on={false}
        />
      </div>
      Release Notes
    </li>
    <li
      key="atd"
      onClick={[Function]}
      style={
        Object {
          "background": "none",
          "color": "#333",
          "cursor": "pointer",
          "paddingLeft": "2.25rem",
        }
      }
    >
      ATS Guide
    </li>
  </ul>
</Fragment>
`;
